app/routes.js

```diff
@@ -1,13 +1,18 @@
 // @flow
 import React from 'react';
-import { Route } from 'react-router';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Switch, Route } from 'react-router';
 import App from './containers/App';
 import HomePage from './containers/HomePage';
 import CounterPage from './containers/CounterPage';
 
-export default (
-  <Route path="/" component={App}>
-    <Route path="/" component={HomePage} />
-    <Route path="/counter" component={CounterPage} />
-  </Route>
+export default () => (
```

Comment:
This now exports a function that returns the Router construct, not sure why it errors if the React.createElement is directly exported (see error below). When loggin the function it will only log once unless something is changed and HMR kicks in. In that case the router function is called again. HMR seems to work as far as I tested.

![screen shot 2017-04-08 at 9 50 28 am](https://cloud.githubusercontent.com/assets/546900/24826944/f5e0b0a0-1c41-11e7-8724-500781c3e3eb.png)